docs/Tutorial_DAMP.ipynb

```diff
@@ -0,0 +1,408 @@
+{
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #98.                    T[start_idx : query_idx + m],
</pre><p>In line #94, we update <code>start_idx</code> as follows:</p><pre class="ql-syntax" spellcheck="false">start_idx = max(0, query_idx + m - prefix) 
</pre><p> </p><p>where prefix is doubled in each iteration. Note that the function MASS computes the distance between the query</p><pre class="ql-syntax" spellcheck="false">Q=T[query_idx : query_idx + m] 
</pre><p>and <code>T[start_idx : query_idx + m]</code> . Therefore, in the next iteration, when <code>start_idx</code> gets lower, we need to compute distance profile for a longer time series, i.e. <code>T[start_idx_lower : query_idx + m]</code></p><p> </p><p>However, this means that we are unnecessary re-computing the distance between <code>Q</code> and the subsequences in <code>T[start_idx : query_idx + m]</code> .</p><p> </p><p>Fig. 8 of the <a href="https://www.cs.ucr.edu/~eamonn/DAMP_long_version.pdf" target="_blank">paper</a> shows the correct way. However, Table 2 seems to not update the stop index.</p><p> </p><p> </p><p> </p><p> </p><p>Side Note:</p><p>The DAMP paper  compute the distance between the query <code>Q</code> and <code>T[start_idx : query_idx]</code>   as they consider one full window size for exlcusion zone. However, in this implementation, I considered to compute the distance upto <code>query_idx + m</code>  but apply <code>core.apply_excl	usion_zone</code>  instead.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/TDAmeritrade/stumpy/pull/872/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

Reply:
> However, Table 2 seems to not update the stop index.

Also, it seems that the authors of `DAMP` in the MATLAB code did consider updating the `stop_index` which is reasonable.